crates/blockifier/src/concurrency/worker_logic.rs

```diff
@@ -30,23 +29,23 @@ pub mod test;
 const EXECUTION_OUTPUTS_UNWRAP_ERROR: &str = "Execution task outputs should not be None.";
 
 #[derive(Debug)]
-pub struct ExecutionTaskOutput {
+pub struct ExecutionTaskOutput<V: VisitedPcs> {
     pub reads: StateMaps,
     pub writes: StateMaps,
     pub contract_classes: ContractClassMapping,
-    pub visited_pcs: HashMap<ClassHash, HashSet<usize>>,
+    pub visited_pcs: V,
     pub result: TransactionExecutionResult<TransactionExecutionInfo>,
 }
 
-pub struct WorkerExecutor<'a, S: StateReader> {
+pub struct WorkerExecutor<'a, S: StateReader, V: VisitedPcs> {
     pub scheduler: Scheduler,
     pub state: ThreadSafeVersionedState<S>,
     pub chunk: &'a [Transaction],
-    pub execution_outputs: Box<[Mutex<Option<ExecutionTaskOutput>>]>,
+    pub execution_outputs: Box<[Mutex<Option<ExecutionTaskOutput<V>>>]>,
     pub block_context: &'a BlockContext,
     pub bouncer: Mutex<&'a mut Bouncer>,
 }
-impl<'a, S: StateReader> WorkerExecutor<'a, S> {
+impl<'a, S: StateReader, V: VisitedPcs + Default + Debug> WorkerExecutor<'a, S, V> {
```

Comment:
Either move `V`'s bounds to a `where` clause or have a type alias if it is used elsewhere.